WcaOnRails/app/models/competition.rb

```diff
@@ -43,6 +43,12 @@ def must_have_at_least_one_event
       errors.add(:eventSpecs, "Competition must have at least one event")
     end
   end
+  validate :must_have_at_least_one_delegate, if: :isConfirmed?
+  def must_have_at_least_one_delegate
+    if delegate_ids.length == 0
+      errors.add(:delegates, "Competition must have at least one WCA delegate")
```

Comment:
The delegates method is a magic method from Rails that looks up all the delegates in the `competition_delegates` table and returns the delegates as an array. See https://github.com/cubing/worldcubeassociation.org/blob/master/WcaOnRails/app/models/competition.rb#L11 for the line of code that tells ActiveRecord to create this delegates method.